packages/vats/src/localchain.js

```diff
@@ -27,33 +31,86 @@ const { Fail } = assert;
  * }} LocalChainPowers
  */
 
+/**
+ * @param {import('@agoric/zone').Zone} zone
+ */
+const prepareDepositPaymentHandler = zone =>
+  zone.exoClass(
+    'DepositPaymentHandler',
+    M.interface('DepositPaymentHandlerI', {
+      onFulfilled: M.call(BrandShape)
+        .optional({
+          bank: M.remotable('Bank Power'),
+          payment: PaymentShape,
+          optAmountShape: M.or(M.undefined(), AmountShape),
+        })
+        .returns(M.promise()),
+    }),
+    () => ({}),
+    /** @type {Watcher<Brand<'nat'>, Amount<'nat'>>} */
```

Comment:
We struggled a little on the return type of this handler (defined as `depositPaymentHandler` further down). Without this declaration - `Watch<T, TResult1>` - it thinks the return type is `T` (the value we get in `onFulfilled`). _Returning the identity type is the default behavior of `watch` when no handler is supplied._

Reply:
As an aside, a "Watcher" is what you're currently calling "Handler", and I think what you're calling "Watcher" is actually just a "promise" or "vow".

Reply:
You're right, this typecast is incorrect here and not needed. I think there were typos in the `depositPaymentHandler` handler so we didn't have inferred types.